### PR TITLE
feat: make buttons inherit stuff

### PR DESCRIPTION
**Pull request by ssikande**
As per Sebastians code:https://github.com/paranext/paranext-core/tree/make-disable-button-inherit-button
Made changes so the buttons also inherit more props

- - -
This change is [Reviewable](https://reviewable.io/reviews/paranext/paranext-core/1035)

### CODE REVIEW

**Review by jolierabideau — CHANGES_REQUESTED** [review]
> Reviewed 11 of 11 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/1035)* status: all files reviewed, 2 unresolved discussions (waiting on @ssikande)

___
*[`lib/platform-bible-react/src/components/advanced-components/extension-marketplace/filter-dropdown.component.tsx` line 37 at r1](https://reviewable.io/reviews/paranext/paranext-core/1035#-O3DsKUl-K7pHsXH-Rlt:-O3DsKUm0dv0Scl3ho6n:b-46ax2h) ([raw file](https://github.com/paranext/paranext-core/blob/b979c9230856c9e63db09acafdc39bb4df31d00a/lib/platform-bible-react/src/components/advanced-components/extension-marketplace/filter-dropdown.component.tsx#L37)):*
> ```XML
> };
> 
> export type FilterDropdownProps = {
> ```

I know the DropdownMenu doesn't have an id included. But can we extend the Dropdown menu props on this as well, like how you did in the buttons? And then keep the id the way you have it right now 
___
*[`lib/platform-bible-react/src/components/advanced-components/extension-marketplace/filter-dropdown.component.tsx` line 54 at r1](https://reviewable.io/reviews/paranext/paranext-core/1035#-O3Ds2kd3ohqqC-P13lJ:-O3Ds2kd3ohqqC-P13lK:bt3g9fa) ([raw file](https://github.com/paranext/paranext-core/blob/b979c9230856c9e63db09acafdc39bb4df31d00a/lib/platform-bible-react/src/components/advanced-components/extension-marketplace/filter-dropdown.component.tsx#L54)):*
> ```XML
> export default function FilterDropdown({ id, groups }: FilterDropdownProps) {
>   return (
>     <div id={id}>
> ```

Interesting that the DropdownMenu doesn't have an `id` on it. Can you add a todo here to remove that when we add an id on the shadcn dropdown menu.

**Review by jolierabideau — CHANGES_REQUESTED** [review]
> Reviewed 6 of 6 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/1035)* status: all files reviewed, 1 unresolved discussion (waiting on @ssikande)

___
*[`lib/platform-bible-react/src/components/advanced-components/extension-marketplace/filter-dropdown.component.tsx` line 37 at r1](https://reviewable.io/reviews/paranext/paranext-core/1035#-O3DsKUl-K7pHsXH-Rlt:-O3ElCWT80s4Uy_KmybN:bxorr8n) ([raw file](https://github.com/paranext/paranext-core/blob/b979c9230856c9e63db09acafdc39bb4df31d00a/lib/platform-bible-react/src/components/advanced-components/extension-marketplace/filter-dropdown.component.tsx#L37)):*
<details><summary><i>Previously, jolierabideau wrote…</i></summary><blockquote>

I know the DropdownMenu doesn't have an id included. But can we extend the Dropdown menu props on this as well, like how you did in the buttons? And then keep the id the way you have it right now
</blockquote></details>

Ohh there aren't props exported for just `DropdownMenu`?  Then let's wait on this, sorry about that. There are different things available on the `DropdownMenu` and `DropdownMenuContent` so I don't want to extend `DropdownMenuContent`

**Review by ssikande — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/1035)* status: 5 of 11 files reviewed, 1 unresolved discussion (waiting on @jolierabideau)

___
*[`lib/platform-bible-react/src/components/advanced-components/extension-marketplace/filter-dropdown.component.tsx` line 37 at r1](https://reviewable.io/reviews/paranext/paranext-core/1035#-O3DsKUl-K7pHsXH-Rlt:-O3Er5e56EOPBNXvINOA:b-896fix) ([raw file](https://github.com/paranext/paranext-core/blob/b979c9230856c9e63db09acafdc39bb4df31d00a/lib/platform-bible-react/src/components/advanced-components/extension-marketplace/filter-dropdown.component.tsx#L37)):*
<details><summary><i>Previously, jolierabideau wrote…</i></summary><blockquote>

Ohh there aren't props exported for just `DropdownMenu`?  Then let's wait on this, sorry about that. There are different things available on the `DropdownMenu` and `DropdownMenuContent` so I don't want to extend `DropdownMenuContent`
</blockquote></details>

Done.

File: lib/platform-bible-react/src/components/advanced-components/extension-marketplace/buttons/enable-button.component.tsx [hunk 1]
```diff
@@ -1,32 +1,44 @@
...
  * visuals for active enabling and idle states.
  *
+ * @param id Optional unique identifier
  * @param isEnabling The enabling boolean value determines the state of the button.
  * @param handleClick The handleClick function is called when the button is clicked.
  * @returns A button that can be used to enable.
  */
-export default function EnableButton({ isEnabling, handleClick }: EnableButtonProps) {
+export default function EnableButton({
+  id,
```
Comment (Sebastian-ubs) [inline lib/platform-bible-react/src/components/advanced-components/extension-marketplace/buttons/enable-button.component.tsx:24]: is there a benefit of mentioning "id" specifically here (and for the other buttons)? It should already come with `props` coming from ButtonProps as an optional parameter, right?

File: lib/platform-bible-react/src/components/advanced-components/extension-marketplace/filter-dropdown.component.tsx [hunk 2]
```diff
@@ -35,46 +35,53 @@ export type DropdownGroup = {
...
-                </div>
-              ))}
-            </DropdownMenuGroup>
-            <DropdownMenuSeparator />
-          </div>
-        ))}
-      </DropdownMenuContent>
-    </DropdownMenu>
+    <div id={id}>
+      {' '}
```
Comment (Sebastian-ubs) [inline lib/platform-bible-react/src/components/advanced-components/extension-marketplace/filter-dropdown.component.tsx:55]: this whitepace string should not be needed

**Review by Sebastian-ubs — CHANGES_REQUESTED** [review]

File: lib/platform-bible-react/src/components/advanced-components/extension-marketplace/buttons/enable-button.component.tsx (hunk 1 above)
Comment (ssikande) [reply lib/platform-bible-react/src/components/advanced-components/extension-marketplace/buttons/enable-button.component.tsx:24]: i added these as per Jolie's comment for testing. Jolie mentiioned that Roopa uses optional ids on the components to test them. But if the ButtonProps already contains an id parameter for testing i can go ahead and remove those changes!

**Review by ssikande — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/1035)* status: all files reviewed, 2 unresolved discussions (waiting on @Sebastian-ubs)

___
*[`lib/platform-bible-react/src/components/advanced-components/extension-marketplace/filter-dropdown.component.tsx` line 55 at r3](https://reviewable.io/reviews/paranext/paranext-core/1035#gh-1701491585:-O3IFqJG9XsBgHiIFQRd:b-896fix) ([raw file](https://github.com/paranext/paranext-core/blob/4624295c1573e8dd8f5bc6a297fc553e1599497e/lib/platform-bible-react/src/components/advanced-components/extension-marketplace/filter-dropdown.component.tsx#L55)):*
<details><summary><i>Previously, Sebastian-ubs wrote…</i></summary><blockquote>

this whitepace string should not be needed
</blockquote></details>

Done.

**Review by ssikande — COMMENTED** [review]
> Reviewed 1 of 11 files at r1, 10 of 10 files at r4, all commit messages.
> Dismissed @Sebastian-ubs from 2 discussions.
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/1035)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved

**Review by Sebastian-ubs — APPROVED** [review]
> Reviewed 1 of 11 files at r1, 10 of 10 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/paranext/paranext-core/1035)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved

**Pull request merged**